utils/rate/gauge.go

```diff
@@ -52,25 +52,25 @@ func (g *Gauge) rateToGauge(valuesSum, calls float64) float64 {
 // Rate1 returns the one-minute moving average of the gauge values.
 // Cannot be larger than max(largest input value, 0)
 func (g *Gauge) Rate1() float64 {
-	return g.rateToGauge(g.input.Rate1(), g.count.Rate1())
+	return g.rateToGauge(g.input.Snapshot().Rate1(), g.count.Snapshot().Rate1())
```

Comment:
This function and the ones below should only take a single snapshot and retrieve the needed information from the same snapshot to be consistent.

Reply:
`g.input` and `g.count` are standalone meters - I cannot obtain a single snapshot for both of them